packages/poc-v2/test/test_StateUtils.ts

```diff
@@ -37,6 +40,30 @@ describe('StateUtils', () => {
     }
   })
 
-
+  it('test payReward() function', async () => {
+    const stakeValues = [
+      // [oldStaked, expectedReward, newStaked]
+      ['10000000000000000000000000', '4807692307692310000000', '10004807692307700000000000'],
+      ['1000000000000000000000000', '913461538461539000000', '1000913461538460000000000'],
+      ['1500000000000000000000000', '2534855769230770000000', '1502534855769230000000000'],
+      ['15000000000000000000000000', '38022836538461500000000', '15038022836538500000000000']
+    ]
+    const roundingError = hre.ethers.BigNumber.from('10000000000000000'); // 10^16
+    for (let [oldStaked, expectedReward, newStaked] of stakeValues) {
+      const oldPoolBalance = await token.balanceOf(pool.address);
+      await pool.testPayReward(hre.ethers.BigNumber.from(oldStaked));
+      // check new total staked value
+      const realStakedNow = await pool.getTotalStakedNow();
+      const newStakedBN = hre.ethers.BigNumber.from(newStaked);
+      const stakeDifference = realStakedNow.sub(newStakedBN).abs();
+      expect(stakeDifference).to.be.lte(roundingError);
+      // check that pool balance increased by expected reward amount
+      const newPoolBalance = await token.balanceOf(pool.address);
+      const balanceDifference = newPoolBalance.sub(oldPoolBalance);
+      const expectedRewardBN = hre.ethers.BigNumber.from(expectedReward);
+      const rewardDifference = balanceDifference.sub(expectedRewardBN).abs();
+      expect(rewardDifference).to.be.lte(roundingError);
+    }
```

Comment:
really good work. same comment RE explicit calculation of expected values, but let's also get on a call tomorrow(?) to walk through this function and all of those calculations